components/Chips/src/MDCChipCollectionViewCell.h

```diff
@@ -0,0 +1,28 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+@class MDCChipView;
+
+/*
+ A collection view cell containing a single MDCChipView.
+ */
+@interface MDCChipCollectionViewCell : UICollectionViewCell
```

Comment:
Can this be private?

Reply:
I don't think so. The intention is for people to create their own UICollectionView and manage the data source themselves.

Reply:
Ah, interesting. Can you include an example of usage in the PR? I'm not able to clearly see how this component is meant to be used.

Reply:
I've pushed up an example that shows one use case.

Reply:
If we remove this cell, I expect that there'd be relatively minor changes to client code in that they'd have to implement their own cell with a chip inside. Given that we're already expecting them to create and manage their own collection view, would it be reasonable for us to remove this cell class so that we have one less API to maintain?

Reply:
I would kind of expect a protocol and we have one class that implements it. Or you can roll your own.

Reply:
@willlarche Who is the intended client of the protocol?

Reply:
@jverkoey whomever is doing the layout calculations. I don't know much about custom layouts but I would have guessed that it would need to know some standard information about what it's laying out since these are dynamically sized. Maybe flow layout does that for you automatically? Or I know you can use autolayout to get size but that requires, often, a cached cell for sizing.

I guess I want some help understanding how complicated we are expecting the layout to be above what comes free. And if it is very complicated, would it be helpful to have a common API, like from a protocol, that you could make the views conform to to give the layout machine the info it needs but still allow anyone to make their own?

Reply:
The cell has a few pieces of logic that are unintuitive and if we expected clients to recreate it every time I suspect they would make a number of common mistakes. For example collection views manage highlighted and selected state for the cells so ideally we pipe that through to the Chip. I don't think providing a Cell is too providing too much API and it would be a mistake to exclude it.

As for a protocol for layout calculations, UICollectionView API already supports everything we need for that. If you want self sizing cells you use self sizing cells, if you want explicitly sized cells you use explicit sizes.